### PR TITLE
GIX-2080: Fix design.spec e2e test

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Enable tokens page in mainnet.

In this PR, fix the design e2e test. The screenshot was taken from `/accounts` instead of the expected `/tokens`.

I have also changed it to use `/` instead of `/tokens`.

# Changes

* Go again to `/` after setting the flag.
* Replace content with XXX before the screenshot.

# Tests

Only test changes.

# Todos

- [ ] Add entry to changelog (if necessary).

Not necessary.

### CODE REVIEW

**Comment by lmuntaner** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #4271** <a href="https://app.graphite.dev/github/pr/dfinity/nns-dapp/4271?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dfinity/nns-dapp/4271?utm_source=stack-comment).

**Comment by lmuntaner** [comment]
@dskloetd please review

File: frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte [hunk 1]
```diff
@@ -89,9 +89,12 @@
   <div role="cell" class={`mobile-row-cell left-cell ${cellAreaName(0)}`}>
     <span class="mobile-only">{$i18n.tokens.balance_header}</span>
     {#if userTokenData.balance instanceof UnavailableTokenAmount}
-      <span data-tid="token-value-label"
-        >{`-/- ${userTokenData.balance.token.symbol}`}</span
-      >
+      <!-- The label is within its own element because it will be replaced in screenshot testing. -->
+      <div data-tid="token-value-label" class="balance-unavailable">
+        <!-- The two spans are needed so that when we get the text from "token-value-label" there is a space between amount and token.  -->
+        <span>{`-/-`}</span>
```
Comment (dskloetd) [inline frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte:95]: You can use normal double quotes now that there are no variables in this string.

File: frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte [hunk 2]
```diff
@@ -89,9 +89,12 @@
   <div role="cell" class={`mobile-row-cell left-cell ${cellAreaName(0)}`}>
     <span class="mobile-only">{$i18n.tokens.balance_header}</span>
     {#if userTokenData.balance instanceof UnavailableTokenAmount}
-      <span data-tid="token-value-label"
-        >{`-/- ${userTokenData.balance.token.symbol}`}</span
-      >
+      <!-- The label is within its own element because it will be replaced in screenshot testing. -->
+      <div data-tid="token-value-label" class="balance-unavailable">
+        <!-- The two spans are needed so that when we get the text from "token-value-label" there is a space between amount and token.  -->
```
Comment (dskloetd) [inline frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte:94]: Can you just use an explicit space similar to line 39 of AmountDisplay? It can probably even be inside `{"-/- "}`, right?

File: frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte [hunk 3]
```diff
@@ -89,9 +89,12 @@
...
+        <span>{`-/-`}</span>
+        <span data-tid="label">{userTokenData.balance.token.symbol}</span>
```
Comment (dskloetd) [inline frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte:96]: Did you mean `symbol` instead of `label`?

File: frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte [hunk 4]
```diff
@@ -235,6 +238,15 @@
     align-items: center;
   }
 
+  .balance-unavailable {
+    // display: inline-flex;
+    // gap: var(--padding-0_5x);
```
Comment (dskloetd) [inline frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte:243]: Why is this here?

File: frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte [hunk 5]
```diff
@@ -235,6 +238,15 @@
...
+    display: inline-grid;
```
Comment (dskloetd) [inline frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte:244]: Do we really need a grid for something so simple?

File: frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte [hunk 6]
```diff
@@ -235,6 +238,15 @@
...
+    grid-template-columns: repeat(2, auto);
+    grid-gap: var(--padding-0_5x);
```
Comment (dskloetd) [inline frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte:246]: Doesn't having the space already give us the distance we need? I feel like I'm missing something.

File: frontend/src/tests/e2e/screenshots/design.spec.ts-Design-Login-1-Google-Chrome-linux.png
Comment (dskloetd) [inline frontend/src/tests/e2e/screenshots/design.spec.ts-Design-Login-1-Google-Chrome-linux.png:1]: Is it really worth going through all this effort to add those few rows, which we then partly block out?
It doesn't look like the extra rows add any extra styling that we potentially might not catch breakages in without them.

Comment (dskloetd) [inline frontend/src/tests/e2e/screenshots/design.spec.ts-Design-Login-1-Google-Chrome-linux.png:1]: I see changes in multiple tests. How come only 1 screenshot needed to be updated?

File: frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte]: Done

File: frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte (hunk 2 above)
Comment (lmuntaner) [reply frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte]: DOne

File: frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte (hunk 4 above)
Comment (lmuntaner) [reply frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte]: It was from a test that I did and forgot to remove. The space does the trick, Therefore, no need for any specific style.

Removed.

File: frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte (hunk 5 above)
Comment (lmuntaner) [reply frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte]: Removed.

File: frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte (hunk 6 above)
Comment (lmuntaner) [reply frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte]: Yes. The space was not in the string itself and the blank space between spans didn't show a space in the html. Only when getting the text from it.

File: frontend/src/tests/e2e/screenshots/design.spec.ts-Design-Login-1-Google-Chrome-linux.png
Comment (lmuntaner) [reply frontend/src/tests/e2e/screenshots/design.spec.ts-Design-Login-1-Google-Chrome-linux.png:1]: That also requires extra effort, no? We would need to remove the rows below it.

Whatever we do, we need to manipulate the DOM.

Ideally we would set the same SNS names.

Comment (dskloetd) [reply frontend/src/tests/e2e/screenshots/design.spec.ts-Design-Login-1-Google-Chrome-linux.png:1]: I'm must be seriously missing something.
In the diff of the screenshot, I see on the left a screenshot without all the extra rows.
That's the screenshot how it is right now, isn't it?
Why is it like that?
Why can't we keep that?

Comment (lmuntaner) [reply frontend/src/tests/e2e/screenshots/design.spec.ts-Design-Login-1-Google-Chrome-linux.png:1]: Ahh, I see. I mentioned in the PR description:

> In this PR, fix the design e2e test. The screenshot was taken from /accounts instead of the expected /tokens.

It was taking a screenshot of `/accounts`, which is only ICP.

Comment (dskloetd) [reply frontend/src/tests/e2e/screenshots/design.spec.ts-Design-Login-1-Google-Chrome-linux.png:1]: Is it important that the screenshot is taken from `/tokens`? Why don't we continue to take it from `/accounts`, given that the only difference doesn't really add important styling that we need to test and is causing a hassle?

Comment (lmuntaner) [reply frontend/src/tests/e2e/screenshots/design.spec.ts-Design-Login-1-Google-Chrome-linux.png:1]: Done.

**Comment by lmuntaner** [comment]
@dskloetd ready for another review.

The screenshot didn't change because it's the same as in `/accounts`.

The changes are to make it more stable by waiting for the proper things and not relying on the redirection.

File: frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte [hunk 7]
```diff
@@ -89,9 +89,13 @@
   <div role="cell" class={`mobile-row-cell left-cell ${cellAreaName(0)}`}>
     <span class="mobile-only">{$i18n.tokens.balance_header}</span>
     {#if userTokenData.balance instanceof UnavailableTokenAmount}
-      <span data-tid="token-value-label"
-        >{`-/- ${userTokenData.balance.token.symbol}`}</span
-      >
+      <!-- The label is within its own element because it will be replaced in screenshot testing. -->
```
Comment (dskloetd) [inline frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte:92]: Is this still true?

File: frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte]: reverted

File: frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte (hunk 2 above)
Comment (lmuntaner) [reply frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte]: reverted

File: frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte (hunk 3 above)
Comment (lmuntaner) [reply frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte]: reverted

File: frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte (hunk 7 above)
Comment (lmuntaner) [reply frontend/src/lib/components/tokens/TokensTable/TokensTableRow.svelte]: No, and I can keep it as it was before. I reverted.

**Comment by lmuntaner** [comment]
@dskloetd ready for another review

**Review by dskloetd — APPROVED** [review]
Thanks!
